recipes/mosquitto/2.x/conanfile.py

```diff
@@ -0,0 +1,82 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+
+class Mosquitto(ConanFile):
+    name = "mosquitto"
+    license = "EPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://mosquitto.org"
+    description = """Eclipse Mosquitto MQTT library, broker and more"""
+    topics = ("MQTT", "IoT", "eclipse")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+                "ssl": [True, False],
+                "clients": [True, False],
+                "broker": [True, False],
+                "apps": [True, False],
+                "plugins": [True, False],
+                "cjson": [True, False],
+            }
+    default_options = {"shared": False,
+                        "ssl": True,
+                        "clients": False,
+                        "broker": False,
+                        "apps": False,
+                        "plugins": False,
+                        "cjson": False
+    }
+
+    _cmake = None
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        if self.options.ssl:
+            self.requires("openssl/1.1.1i")
+        if self.options.cjson:
+            self.requires("cjson/1.7.14")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.replace("-", ".") + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["WITH_TLS"] = self.options.ssl
+        self._cmake.definitions["WITH_CLIENTS"] = self.options.clients
+        self._cmake.definitions["WITH_BROKER"] = self.options.broker
+        self._cmake.definitions["WITH_APPS"] = self.options.apps
+        self._cmake.definitions["WITH_PLUGINS"] = self.options.plugins
+        self._cmake.definitions["DOCUMENTATION"] = False
+        if self.options.ssl:
+            self._cmake.definitions["OPENSSL_SEARCH_PATH"] = self.deps_cpp_info["openssl"].rootpath
+            self._cmake.definitions["OPENSSL_ROOT_DIR"] = self.deps_cpp_info["openssl"].rootpath
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("edl-v10", src=self._source_subfolder, dst="licenses")
+        self.copy("edl-v20", src=self._source_subfolder, dst="licenses")
+        self.copy("LICENSE.txt", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib","pkgconfig"))
+
+    def package_info(self):
+        #self.cpp_info.libdirs = ["lib"] # thats default anyway
```

Comment:
Can you please set `self.cpp_info.names["pkg_config"]`?

Reply:
are the pkgconfilg files useful? since the prefix in them is usually an absolute path, I thought the can not be used by conan

Reply:
These can be generated by the `pkg_config` generator.

Reply:
this makes the test package fail

```
ERROR: mosquitto/2.0.3: Error in package_info() method, line 87
	self.cpp_info.names["pkg_config"]
	KeyError: 'pkg_config'
```
and still triggers this hook error
```
ERROR: [PC-FILES (KB-H020)] Found files: ./lib/pkgconfig/libmosquitto.pc; ./lib/pkgconfig/libmosquittopp.pc (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H020) 
```

so I do not think I can do that

Reply:
You need to remove the `lib/pkgconfig` folder,
but by settings `self.cpp_info.names["pkg_config"] = "mqtt"` (or whatever name),
you instruct conan to, when adding `pkg_config` to `generators`, to create a `mqtt.pc` file in the conan install folder.

Reply:
good to know, thanks
but there are 2 pkgconfil files, can this be used 2 times

Reply:
Only if you're using components and set its `libs` attribute accordingly.